src/Provider/PaymentRedirectionProvider.php

```diff
@@ -59,67 +58,31 @@ public function __construct(Context $context, $moduleName, SaferPayFieldReposito
      */
     public function provideRedirectionLinkByPaymentMethod($paymentMethod)
     {
-        $redirectionLink = $this->context->link->getModuleLink(
-            $this->moduleName,
-            'validation',
-            ['saved_card_method' => $paymentMethod, SaferPayConfig::IS_BUSINESS_LICENCE => false],
-            true
-        );
+        $paymentType = $this->paymentTypeProvider->get($paymentMethod);
 
-        if ($this->isIframeRedirect($paymentMethod)) {
-            $redirectionLink = $this->context->link->getModuleLink(
+        if ($paymentType === PaymentType::HOSTED_IFRAME) {
+            return $this->context->link->getModuleLink(
                 $this->moduleName,
-                'iframe',
+                'hostedIframe',
```

Comment:
can be enums, no ? Or atleast constants

Reply:
It's a  controleller name, not sure how we proceeded with that

Reply:
Pretty sure we just had it as string, but i guess i could create an enum for controllers

Reply:
yeah would be nice so newcommers would understand that it is somehow connected with the controller